### ISSUE TITLE
the genenrated target views become blurry during training

### ISSUE THREAD

**Issue by fangchuan** [kxhit/EscherNet#17]
Hi @fradif96 
We don't have new modules for cross/self-attention. It's the same attention layers but just reshape the latent features from ((b t) l d) -> (b (t l) d) [here](https://github.com/kxhit/EscherNet/blob/cf6b220e5085acfad1d20d68afde3e101e2af1e2/6DoF/diffusers/models/attention_processor.py#L1028). The trainable layers are UNet, ConvNext image encoder [here ](https://github.com/kxhit/EscherNet/blob/cf6b220e5085acfad1d20d68afde3e101e2af1e2/6DoF/train_eschernet.py#L626)(we didn't use CLIP as it only captures high-level semantics), the VAE is frozen, and we fintuned from SD1.5 inited [here](https://github.com/kxhit/EscherNet/blob/cf6b220e5085acfad1d20d68afde3e101e2af1e2/6DoF/train_eschernet.py#L553).

Hope it helps and please let me know if something is still unclear.

_Originally posted by @kxhit in https://github.com/kxhit/EscherNet/issues/1#issuecomment-1935166434_

**Comment by fangchuan** [comment]
Dear author, I finetune EscherNet (load the pretrained weights of 6DoF) on an indoor scene dataset which contains multiview images with known poses.
By looking through the tensorflow/wandb monitor, the training process is fine at the beginning. But the results on validation set become blurry (red circle areas in the figure) after about 20K steps, I am quite curious about the reason, do you met similar problem during training? 
<img width="963" alt="image" src="https://github.com/user-attachments/assets/17e8b420-72b7-43bd-b3e5-4204888b6364">
<img width="963" alt="image" src="https://github.com/user-attachments/assets/cb4b95c7-9a72-454e-9885-ad51a71dd826">


if the number of input views equal 8, the blurry problem is alleviated in some extent, but the generated images still fails to preserve the texture of inputs.  
<img width="965" alt="image" src="https://github.com/user-attachments/assets/e1a651e1-d3e7-4dc0-9c4c-3e37969f6cc7">

I did not revise the training pipeline, optimize the unet and image_encoder follow EscherNet, could you help me or give me some clue about the problem of texture? @kxhit

Many thanks!

**Comment by kxhit** [comment]
Hi, it looks like an overfitting problem. How's the visualisation of the training split and the PSNR metrics on train/val splits?
It makes sense that the more views are given, the better results. But it is weird if the validation gets blurry/worse.

**Comment by fangchuan** [comment]
> Hi, it looks like an overfitting problem. How's the visualisation of the training split and the PSNR metrics on train/val splits? It makes sense that the more views are given, the better results. But it is weird if the validation gets blurry/worse.

The figures above are on validation set, results on training set is illustrated as following, it still exhibit severe blurry if fewer inputs are given. More input views get better pose controlling and texture perseverance, but the blurry problem still exist, the overall PSNR of generated target views is 14, for 8 inputs. 
![image](https://github.com/user-attachments/assets/bef9869f-5cb4-4c43-80cc-79db58a61d8c)
![image](https://github.com/user-attachments/assets/2019ff10-4acd-40d1-b557-682ae485dd46)
![image](https://github.com/user-attachments/assets/63f450fa-e9d4-40aa-913c-9b58ed6b3441)


Does the training resolution matters? I use 8 input views with 256x256 results, and 8 target views at same size.

**Comment by fangchuan** [comment]
> Hi, it looks like an overfitting problem. How's the visualisation of the training split and the PSNR metrics on train/val splits? It makes sense that the more views are given, the better results. But it is weird if the validation gets blurry/worse.

Dear author @kxhit , I have another question. Why choose the ConvNextV2 rather than the backbone UNet  in the way like ControlNet? It seems the downsample block of UNet could preserve more accurate texture of input images, which has been proven by [Genwarp](https://arxiv.org/pdf/2405.17251).

In my experiments, I found that the testing results on validation set still fail to preserve some details of the inputs, even increasing the input images to 8. I want to discuss with you about what kind of image encoder would be a better choice in scene-level multi-view generation ? 

Again, Thanks for your time!

**Comment by kxhit** [comment]
@fangchuan
We chose tiny ConvNextV2 just because it is small enough to fit into GPU memory, as we want to keep batch size big enough for stable training. It can be swapped by any more powerful image encoder, or even simply encoded by UNet, I think it is definitely worth a try if you have enough computing.
